AudioLoader/Music.py

```diff
@@ -697,3 +702,145 @@ def available_groups(self, group):
+        self.sr = 44100
+        self.segment = int(self.seq_duration * self.sr)
+        self.split = split
+        self.samples_per_track = samples_per_track
+        self.random_track_mix = random_track_mix
+        self.random = random
+        self.sources = ['drums', 'bass', 'other', 'vocals']
+
+        self.transform = transform
+
+        setup_path = os.path.join(
+            musdb.__path__[0], 'configs', 'mus.yaml'
+        )
+        with open(setup_path, 'r') as f:
+            self.setup = yaml.safe_load(f)
+
+        self.tracks, self.track_lenghts = self.load_mus_tracks(
+            self.sr, self.subsets, self.split)
+
+        if self.seq_duration <= 0:
+            self._size = len(self.tracks)
+        elif self.random:
+            self._size = len(self.tracks) * self.samples_per_track
+        else:
+            chunks = [l // self.segment for l in self.track_lenghts]
+            cum_chunks = np.cumsum(chunks)
+            self.cum_chunks = cum_chunks
+            self._size = cum_chunks[-1]
+
+    def load_mus_tracks(self, sr, subsets=None, split=None):
+        if subsets is not None:
+            if isinstance(subsets, str):
+                subsets = [subsets]
+        else:
+            subsets = ['train', 'test']
+
+        if subsets != ['train'] and split is not None:
+            raise RuntimeError(
+                "Subset has to set to `train` when split is used")
+
+        print("Gathering files ...")
+        tracks = []
+        track_lengths = []
+        for subset in subsets:
+            subset_folder = os.path.join(self.root, subset)
+            for _, folders, _ in tqdm(os.walk(subset_folder)):
+                # parse pcm tracks and sort by name
+                for track_name in sorted(folders):
+                    if subset == 'train':
+                        if split == 'train' and track_name in self.setup['validation_tracks']:
+                            continue
+                        elif split == 'valid' and track_name not in self.setup['validation_tracks']:
+                            continue
+
+                    track_folder = os.path.join(subset_folder, track_name)
+                    # add track to list of tracks
+                    tracks.append(track_folder)
+
+                    meta = torchaudio.info(os.path.join(
+                        track_folder, 'mixture.wav'))
+                    assert meta.sample_rate == sr
+                    track_lengths.append(meta.num_frames)
+
+        return tracks, track_lengths
+
+    def __len__(self):
+        return self._size
+
+    def _get_random_track_idx(self):
+        return random.randrange(len(self.tracks))
+
+    def _get_random_start(self, length):
+        return random.randrange(length - self.segment + 1)
+
+    def _get_track_from_chunk(self, index):
+        track_idx = np.digitize(index, self.cum_chunks)
+        if track_idx > 0:
+            chunk_start = (index - self.cum_chunks[track_idx]) * self.segment
```

Comment:
When `track_idx`>0, `chunk_start` is always negative.
Maybe `(self.cum_chunks[track_idx] - index) * self.segment` is the correct one?

Reply:
```suggestion
            chunk_start = (index - self.cum_chunks[track_idx-1]) * self.segment
```